### PR TITLE
Py did vs str did

### PR DESCRIPTION
**Pull request by sklump**
Note that throughout aca-py, `did` is typically a string. This gave rise to some ambiguity in passing public (PyDID) DID objects into what expected a string.

Also refactored the DIDInfo duplication away.

### CODE REVIEW

File: aries_cloudagent/resolver/base.py [hunk 1]
```diff
@@ -61,19 +61,17 @@ def supports(self, method: str) -> bool:
...
 
-        did = str(did)
-        did_document = await self._resolve(profile, did)
+        did_document = await self._resolve(profile, str(py_did))
         result = DIDDocument.deserialize(did_document)
         return result
 
     @abstractmethod
-    async def _resolve(self, profile: Profile, did: DID) -> dict:
+    async def _resolve(self, profile: Profile, py_did: DID) -> dict:
```
Comment (dbluhm) [inline aries_cloudagent/resolver/base.py:76]: This line is actually a mistake; the type for `did` should be `str` with the intention being that resolver plugins need not concern themselves with PyDID if they don't want to. Sorry for the mix-up.

File: aries_cloudagent/resolver/default/indy.py [hunk 2]
```diff
@@ -37,21 +37,20 @@ def supported_methods(self) -> Sequence[str]:
         """Return supported methods of Indy DID Resolver."""
         return ["sov"]
 
-    async def _resolve(self, profile: Profile, did: str) -> dict:
+    async def _resolve(self, profile: Profile, py_did: DID) -> dict:
```
Comment (dbluhm) [inline aries_cloudagent/resolver/default/indy.py:40]: See previous comment for this method.

**Review by dbluhm — CHANGES_REQUESTED** [review]
It seems we've triggered some confusion on the `_resolve` method parameter types -- sorry about that. I actually had a fix locally that I just hadn't opened a PR for yet.

File: aries_cloudagent/resolver/base.py (hunk 1 above)
Comment (sklump) [reply aries_cloudagent/resolver/base.py:76]: It's no trouble, let's just converge on what's right

File: aries_cloudagent/resolver/default/indy.py (hunk 2 above)
Comment (sklump) [reply aries_cloudagent/resolver/default/indy.py]: How about now? Better?

**Review by dbluhm — APPROVED** [review]

**Review by andrewwhitehead — APPROVED** [review]

**Pull request merged**